### ISSUE TITLE
Support for signed commits

### ISSUE THREAD

**Issue by mu88** [versionize/versionize#124]
Hey 👋🏻

I've configured all my GitHub repos to only accept signed Git commits. Hence, I've also configured my local Git client accordingly:
```config
[commit]
	gpgsign = true
[user]
	name = mu88
	email = <<the email address>>
	signingkey = <<the key>>
```

This works fine with all Git commands (either via shell or IDE, e. g. PowerShell, Rider, etc.). Here's one example ([link to commit](https://github.com/mu88/ThisIsYourLife/commit/a0d608da83bd13bfc1d2d294a2330a956947369e)):

![image](https://github.com/versionize/versionize/assets/4560672/a6fae46e-271e-430b-a44c-2e98dcab8396)

However, when running `versionize`, Git doesn't require me to provide my password to sign the commit and therefore the commit is unsigned ([link to commit](https://github.com/mu88/ThisIsYourLife/commit/4fc5a4a449849302976a221792fb860fae177d88)):

![image](https://github.com/versionize/versionize/assets/4560672/4f5417c0-a9bf-441b-9ffe-d0a63b09ccf5)

I assume that the used Git library `libgit2sharp` somehow bypasses my local Git config 🤔

So long story short 🙂 how can I create signed commits using `versionize`?

**Comment by cabauman** [comment]
This seems like a nice feature, but after searching around it doesn't seem like libgit2sharp supports this. [Comment from October 2022](https://github.com/cbcrouse/Versioning.NET/issues/34#issuecomment-1288308157):

> Signing commits appears to https://github.com/libgit2/libgit2sharp/issues/1101 by LibGit2Sharp in any convenient way

We can keep an eye on it, but for now I don't see a way.

**Comment by mu88** [comment]
Ouh that's a problem for me :/ but thx anyway for digging into it!

**Comment by mu88** [comment]
As a workaround, I've added the following alias to my Git config:
```
[alias]
    versionize = "!f() { \
        dotnet versionize; \
        original_commit=$(git rev-parse HEAD); \
        tag_name=$(git tag --points-at $original_commit); \
        if [ -z \"$tag_name\" ]; then \
            echo \"No tag found pointing to the original commit.\"; \
            exit 1; \
        fi; \
        echo \"Found tag: $tag_name\"; \
        git tag -d $tag_name; \
        git commit --amend --no-edit -S; \
        new_commit=$(git rev-parse HEAD); \
        git tag $tag_name $new_commit; \
        echo \"Tag $tag_name moved to new commit $new_commit.\"; \
    }; f"
```

This way, I can call `git versionize` which does the following:
1. Run `dotnet versionize`
2. Determine created Git tag
3. Remove Git tag from current commit
4. Sign the commit using `git amend` as it's the most latest commit
5. Add the previous Git tag again

**Comment by cabauman** [comment]
Thanks for sharing a workaround!

**Comment by cabauman** [comment]
Does your workflow depend on gpg signed tags too?

**Comment by mu88** [comment]
I think that's actually a good idea to let the installed Git client handle the delicate GPG dance. Before digging into this issue, I naturally expected that Versionize would use the installed Git client instead of a Git library - but of course, now with a bit more thinking about the latter makes much more sense :)

> Does your dev flow depend on gpg signed tags too?

Can you please elaborate on what you exactly mean? I don't yet get it :)

**Comment by cabauman** [comment]
> I've configured all my GitHub repos to only accept signed Git commits.

I was just curious if this applies to tags, as well. Since gpg also supports signing tags, I may try to use the git client for creating tags too.

**Comment by mu88** [comment]
Ah, I didn't know that 💡 interesting! As you can read from my surprise, I'm not using this feature yet 😉

**Comment by mu88** [comment]
works like a charm, thank you 👏🏻
